### PR TITLE
Add support for design system for changelog

### PR DESCRIPTION
**Pull request by SudharakaP**
This adds support for design system for changelog

Fixes https://github.com/opencollective/opencollective/issues/2813

### CODE REVIEW

**Comment by SudharakaP** [comment]
@Betree : I have converted all the components to use the new styles, but there's one problem I couldn't solve yet. The styled-system colors used by the `StyledTag` doesn't seem to be translated correctly by default. I mean when I use say, 

`<StyledWeekLabel type="error">.....</StyledWeekLabel>` 

it seems I get, 

image

I was wondering maybe rather than using strings we need to use `themeGet` explicitly in the frontend side such as (for [TAG_TYPE_VARIANTS](https://github.com/opencollective/opencollective-frontend/blob/380b5882bbeca74ce4367209ba6b8ffdbfe7ddd7/components/StyledTag.js#L21-L25));

```
const TAG_TYPE_VARIANTS = {
  white: {
    backgroundColor: themeGet('white.full'),
    borderColor: themeGet('colors.black.200'),
  },
```
WDYT? Do you have any idea about this? :thinking:

**Comment by Betree** [comment]
@SudharakaP I think we need to add https://bit.dev/opencollective/design-system/theme and a `ThemeProvider` to the `_app` for it to work.

**Comment by SudharakaP** [comment]
> @SudharakaP I think we need to add https://bit.dev/opencollective/design-system/theme and a `ThemeProvider` to the `_app` for it to work.

@Betree : Thanks much. Yes that was the issue. I've fixed it and think that I've completed everything here. I've change some things and tried to minimize any custom styles we use. Please take a look and if anything needs to change let me know. 

https://changelog-27wexni5x.vercel.app/

File: pages/index.js [hunk 1]
```diff
@@ -58,14 +76,14 @@ export default class Index extends React.Component {
     switch (label.name) {
       case "bug":
         label.name = "fixed";
-        label.color = "0066d6";
+        label.type = "info";
```
Comment (Betree) [inline pages/index.js:79]:
```suggestion
        label.type = "error";
```

File: pages/index.js [hunk 2]
```diff
@@ -58,14 +76,14 @@ export default class Index extends React.Component {
...
         break;
       case "enhancement":
         label.name = "improved";
-        label.color = "e1663f";
+        label.type = "error";
```
Comment (Betree) [inline pages/index.js:83]:
```suggestion
        label.type = "info";
```

**Comment by SudharakaP** [comment]
> Looks good to me, I would just switch `bug` and `enhancement` colors. In my perception red is usually associated with bugs, and it's an important thing ("danger"), while an enhancement is just an info.

@Betree : Sure, that's better. I've made the changes. :+1:

**Comment by Betree** [comment]
Thank you! This is going live :rocket: 

Now that we have the design system implemented in this project, it will be easier to implement UI enhancements and improve the design consistency.

**Comment by SudharakaP** [comment]
> Thank you! This is going live rocket
> 
> Now that we have the design system implemented in this project, it will be easier to implement UI enhancements and improve the design consistency.

You are welcome. :smile:

**Pull request merged**